src/containers/detail/detail.container.tsx

```diff
@@ -0,0 +1,103 @@
+import { connect } from 'react-redux';
+import { StatusTag } from '../../common/components/status-tag/status-tag.component';
+import { SurveyForm, SurveyFormProps } from '../../common/components/survey-form/survey.form';
+import { Survey } from '../../common/types/survey.type';
+import { AppState } from '../../state';
+import { BrandingForm, BrandingFormProps } from './branding.form';
+import './detail.container.less';
+
+export interface DetailContainerProps {
+    survey: Survey;
+    modalVisible: boolean;
+    generalForm: ComponentClass<SurveyFormProps>;
+    brandingForm: ComponentClass<BrandingFormProps>;
+}
+
+const mapStateToProps = (state: AppState): Partial<DetailContainerProps> => ({
+    survey: state.detailState.survey,
+});
+
+// @ts-ignore
+@connect(mapStateToProps)
+export class DetailContainer extends Component<DetailContainerProps> {
+    public static defaultProps = {
+        modalVisible: false,
+        generalForm: Form.create()(SurveyForm),
+        brandingForm: Form.create()(BrandingForm),
+    };
+
+    saveGeneral = () => {
+        this.setState({ ...this.state, modalVisible: false });
+        console.log(this.props.generalForm);
+    };
+
+    render() {
+        const { survey, modalVisible } = this.props;
+
+        // TODO: show actual data in these forms
+        const generalFormElement = React.createElement(this.props.generalForm);
+        const brandingFormElement = React.createElement(this.props.brandingForm);
+
+        return (
+            <>
+                {/* TODO: modal state doesn't toggle */}
+                {/* TODO: disable save button when input is not valid (see BrandingForm) */}
+                <Modal
+                    title="Edit"
+                    visible={modalVisible}
+                    onCancel={() => this.setState({ ...this.state, modalVisible: false })}
+                    onOk={() => this.saveGeneral()}
+                    okText="Save"
+                    cancelText="Cancel"
+                >
+                    {generalFormElement}
+                </Modal>
+
+                <h2>
+                    {survey.name}{' '}
+                    <Button
+                        shape="circle"
+                        icon="edit"
+                        onClick={() => this.setState({ ...this.state, modalVisible: true })}
+                    />
+                </h2>
+
+                <Card
+                    title="General"
+                    extra={
+                        <>
+                            <Button>Survey Builder</Button> <Button>Start surveying</Button>
+                        </>
+                    }
+                >
+                    <p>{survey.description}</p>
+
+                    <Button style={{ float: 'right' }}>See results</Button>
+
+                    <ul style={{ listStyleType: 'none' }}>
+                        <li>
+                            {/* TODO: use extended status tag when it's finished */}
```

Comment:
@decrn You were going to add this

Reply:
Yes! Will add tag with icon and name, in two sizes.

Reply:
Update: #43 is in production.

Reply:
I have added the StatusTag as requested.

![image](https://user-images.githubusercontent.com/15365152/52536328-5ef58a00-2d59-11e9-8ac1-a91ced07efd1.png)